staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -150,25 +161,59 @@ func TestWatchErrResultNotBlockAfterCancel(t *testing.T) {
 	wg.Wait()
 }
 
-// TestWatchErrorWhenNoNewFunc checks if an error
-// will be returned when establishing a watch
-// with progressNotify options set
-// when newFunc wasn't provided
-func TestWatchErrorWhenNoNewFunc(t *testing.T) {
-	origCtx, store, _ := testSetup(t, func(opts *setupOptions) { opts.newFunc = nil })
-	ctx, cancel := context.WithCancel(origCtx)
-	defer cancel()
-
-	w, err := store.watcher.Watch(ctx, "/abc", 0, storage.ListOptions{ProgressNotify: true})
-	if err == nil {
-		t.Fatalf("expected an error but got none")
-	}
-	if w != nil {
-		t.Fatalf("didn't expect a watcher because progress notifications cannot be delivered for a watcher without newFunc")
+// TestWatchErrorIncorrectConfiguration checks if an error
+// will be returned when the storage hasn't been properly
+// initialised for watch requests
+func TestWatchErrorIncorrectConfiguration(t *testing.T) {
```

Comment:
This PR is growing too much now making it harder to review.
Can you split this test refactoring to a separate one (without any other changes) - just initially different errors will be returned [and we will change errors later].

Reply:
ptal - https://github.com/kubernetes/kubernetes/pull/120747